smarts/env/opendrive_env.py

```diff
@@ -0,0 +1,207 @@
+
+
+class OpenDriveEnv(gym.Env):
+    """A complete gym environment that wraps a SMARTS simulation for OpenDRIVE scenarios
+    without SUMO traffic and zoo social agents.
+
+    Args:
+        scenarios:
+            a list of directories of the OpenDRIVE scenarios that will be run
+        agent_specs:
+            a list of agents that will run in the environment
+        sim_name:
+            a string that gives this simulation a name
+        headless:
+            true|false envision disabled
+        visdom:
+            true|false visdom integration
+        fixed_timestep_sec:
+            the step length for all components of the simulation
+            (may be None if time deltas are externally-driven)
+        seed:
+            the seed for random number generation
+        envision_endpoint:
+            used to specify envision's url
+        envision_record_data_replay_path:
+            used to specify envision's data replay output directory
+    """
+
+    metadata = {"render.modes": ["human"]}
+    """Metadata for gym's use"""
+
+    def __init__(
+        self,
+        scenarios: Sequence[str],
+        agent_specs: Dict[str, AgentSpec],
+        sim_name=None,
+        shuffle_scenarios=True,
+        headless=False,
+        visdom=False,
+        fixed_timestep_sec=None,
+        seed=42,
+        envision_endpoint=None,
+        envision_record_data_replay_path=None,
+        timestep_sec=None,  # for backwards compatibility (deprecated)
+    ):
+        self._log = logging.getLogger(self.__class__.__name__)
+        self.seed(seed)
+
+        if timestep_sec and not fixed_timestep_sec:
+            warnings.warn(
+                "timestep_sec has been deprecated in favor of fixed_timestep_sec.  Please update your code.",
+                category=DeprecationWarning,
+            )
+        if not fixed_timestep_sec:
+            fixed_timestep_sec = timestep_sec or 0.1
+
+        self._agent_specs = agent_specs
+        self._dones_registered = 0
+
+        self._scenarios_iterator = Scenario.scenario_variations(
+            scenarios,
+            list(agent_specs.keys()),
+            shuffle_scenarios,
+        )
+
+        agent_interfaces = {
+            agent_id: agent.interface for agent_id, agent in agent_specs.items()
+        }
+
+        envision_client = None
+        if not headless or envision_record_data_replay_path:
+            envision_client = Envision(
+                endpoint=envision_endpoint,
+                sim_name=sim_name,
+                output_dir=envision_record_data_replay_path,
+                headless=headless,
+            )
+
+        visdom_client = None
+        if visdom:
+            visdom_client = VisdomClient()
+
+        self._smarts = SMARTS(
+            agent_interfaces=agent_interfaces,
+            traffic_sim=None,
+            envision=envision_client,
+            visdom=visdom_client,
+            fixed_timestep_sec=fixed_timestep_sec,
+        )
```

Comment:
Looking at it, the only major change I see between this env and `HiWayEnv` is the traffic simulator. I could suggest we detect the version of scenario we are using instead and decide if to add the traffic.

Reply:
I don't think it will be a good idea to do so. `opendrive_road_network.py` does not support `SumoTrafficSimulation` but we still be importing it creating a dependency. If we do conditional imports based on scenario version there's still the case that we have a mix of OpenDRIVE and SUMO scenarios. I can add the change you suggested above, but it will look a bit messy.